src/Core/Http/SecureHttpClient.php

```diff
@@ -230,7 +230,7 @@ public function getJWKThumbprint()
      *
      * @return array|string Array of parsed JSON if $returnJson = true, string otherwise
      */
-    public function signedKidRequest($method, $endpoint, $account, array $payload = [], $returnJson = true)
+    public function signedKidRequest($method, $endpoint, $account, ?array $payload = [], $returnJson = true)
```

Comment:
The signature issue with php < 7 can be solve by switching to default value = `null`.
But this would be a bc break. 

What's about:
```diff
 public function signedKidRequest($method, $endpoint, $account, array $payload = [], $returnJson = true) {
+   @trigger_error('method signedKidRequest is deprecated, use signedRequest instead', E_USER_DEPRECATED);
+   return $this->signedRequest($method, $endpoint, $account, $payload, $returnJson);
+}
+public function signedRequest($method, $endpoint, $account, array $payload = null, $returnJson = true) {
```
and updating all calls to `->signedKidRequest` to `signedRequest` and adding `[]` in the 4th arguments when the method is `POST` and the 4th argument were not defined.

Reply:
signedRequest() already exists, with a slightly different body. Advice?

https://github.com/acmephp/acmephp/blob/c3241275593b634a4295fd99e960575932a4af6c/src/Core/Http/SecureHttpClient.php#L111-L139

Here's the payload from signedKidRequest for comparison
```php
$protected = [
            'alg' => $alg,
            'kid' => $account,
            'nonce' => $this->getNonce(),
            'url' => $endpoint,
        ];
```

Reply:
indeed signedRequest already exists and is another way to sign request... 

Reading https://community.letsencrypt.org/t/acme-breaking-change-most-gets-become-posts/71025
> Clients will need to change GET requests into signed POST requests.

The change is maybe a little be bigger, we need to deprecate unsignedRequest. Deprecate argument "method" which will always be "POST", ...

I suggest to deprecate unsignedRequest, signedRequest and signedKidRequest => replace by a single method request.
```php
public function request($endpoint, $account = null, array $data = null, $returnJson = true)

// $client->signedKidRequest
$client->request($endpoint, $account); 
// $client->signedRequest
$client->request($endpoint, null); 
```
or 
```php
public function request($endpoint, $signedPayload, $returnJson = true)

// $client->signedKidRequest
$client->request($endpoint, $client->signKidPayload($endpoint, $account)); 
// $client->signedRequest
$client->request($endpoint, $client->signJwkPayload($endpoint)); 
```
The second one is "cleaner" but need more work in clients. For instance
```diff
- return $this->getHttpClient()->signedKidRequest('POST', $this->getResourceAccount(), $this->getResourceAccount());
+ return $this->getHttpClient()->request($this->getResourceAccount(), $this->getHttpClient()->signKidPayload($this->getResourceAccount(), $this->getResourceAccount(), []));

  // or to avoid dupplicated variables
+ $client = this->getHttpClient();
+ $account = $this->getResourceAccount();
+ return $client->request($account, $client->signKidPayload($account, $account, []));
```

Whats do you think? Tell me if you need help.

Reply:
We still need non-POST requests for getResourceUrl() and getNonce(). Keep `unsignedRequest()` will allows a method and any `request()` is signed that implies POST method? How about these signatures?

```php
public function request($endpoint, array $data = null, $returnJson = true);
public function unsignedRequest($method, $endpoint, array $data = null, $returnJson = true);
public function signKidRequest($endpoint, $account, array $payload = null, $returnJson = true);
public function signJwkRequest($endpoint, array $payload = [], $returnJson = true);
```

and deprecate `signedKidRequest()` + `signedRequest()`.

Slammed with work, have a firm Dec 15 deadline on a deliverable, so my time is a little limited on constructive feedback.

Reply:
could you have a lok to #183 ?

Reply:
LGTM 👍